src/containers/WelcomePage/components/Revision.tsx

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2023-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { useTranslation } from 'react-i18next';
+import { memo, useMemo, useState } from 'react';
+import { IUserData } from '@ndla/types-draft-api';
+import { Alarm } from '@ndla/icons/common';
+import addYears from 'date-fns/addYears';
+import { Select, Option, SingleValue } from '@ndla/select';
+import {
+  ControlWrapperDashboard,
+  DropdownWrapper,
+  StyledDashboardInfo,
+  StyledLink,
+  StyledTopRowDashboardInfo,
+} from '../styles';
+import TableComponent, { FieldElement, TitleElement } from './TableComponent';
+import TableTitle from './TableTitle';
+import formatDate, { formatDateForBackend } from '../../../util/formatDate';
+import { toEditArticle } from '../../../util/routeHelpers';
+import { useSearch } from '../../../modules/search/searchQueries';
+import { getExpirationDate } from '../../ArticlePage/articleTransformers';
+import GoToSearch from './GoToSearch';
+
+interface Props {
+  userData: IUserData | undefined;
+  favoriteSubjects: Option[];
+  ndlaId: string | undefined;
+}
+
+const Revision = ({ userData, favoriteSubjects, ndlaId }: Props) => {
+  const [filterSubject, setFilterSubject] = useState<SingleValue | undefined>(undefined);
+  const [sortOption, setSortOption] = useState<string>('-revisionDate');
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  const { t } = useTranslation();
+
+  const tableTitles: TitleElement[] = [
+    { title: t('form.article.label'), sortableField: 'title' },
+    { title: t('welcomePage.workList.status') },
+    { title: t('welcomePage.workList.primarySubject') },
+    { title: t('welcomePage.revisionDate'), sortableField: 'revisionDate' },
+  ];
+
+  const currentDateAddYear = formatDateForBackend(addYears(new Date(), 1));
+
+  const { data, isInitialLoading } = useSearch(
+    {
+      subjects: userData?.favoriteSubjects!.join(','),
```

Comment:
Kunne vi slått sammen denne og filterSubject lenger nede på en måte? Slik at filterSubject velges dersom den er satt, hvis ikke faller vi tilbake til favoriteSubjects.

Reply:
Kunne oppdatert til
`  {
      subjects: filterSubject ? filterSubject.value : userData?.favoriteSubjects!.join(','),
      'revision-date-to': currentDateAddYear,
      sort: sortOption,
 },` feks?👌

Reply:
Gjør det